central/image/service/service_impl.go

```diff
@@ -296,6 +297,38 @@ func (s *serviceImpl) ScanImageInternal(ctx context.Context, request *v1.ScanIma
 	return internalScanRespFromImage(img), nil
 }
 
+// updateImageFromRequest will update the name of existing image with the one from the request
+// if the names differ and the metadata for the existing image was unable to be pulled previously.
+func updateImageFromRequest(existingImg *storage.Image, reqImgName *storage.ImageName) {
```

Comment:
Hmm, I wonder whether we would also need to update the `Names` field.

I'd expect in the case we first attempt to enrich the image without the correct registry / repository being set that the initial image would fail enrichment and have the name set to `docker.io/ubi@sha256:1` and the names field containing `docker.io/ubi@sha256:1`. 

In this case I'd assume that we want this names field to either be overwritten completely or adjusted as well to reflect the name, otherwise we would get after the successful second enrichment request an image with the name `registry.access.redhat.com/ubi@sha256:1` and a names field containing `docker.io/ubi@sha256:1,registry.access.redhat.com/ubi@sha256:1`.

However, not sure how _prevalent_ this case really is, since AFAICT it would require a digest reference of the initial image we attempt to enrich; and the fact that we don't populate the image names within the UI - but ideally we'd be consistent here.

Reply:
If a digest (instead of tag) is provided as part of the pod spec, then your absolutely correct that the `Names` field will contain both entries - I sent you link to the design doc and extended testing results (via slack) to back this up.  

> In this case I'd assume that we want this names field to either be overwritten completely or adjusted as well to reflect the name, ...

I did consider this and left it as is thinking there was little harm and there is currently no way of knowing if an existing entry in the Names field there due to this use case, or a bad image reference, a valid reference but central lacking an image integration (at the time), temporary registry errors, network errors, etc.  It's possible that `Names` represents a valid reference from some other cluster.

The image service will only update the DB if an enrichment succeeds when there is an existing entry the DB.   Combining that with the `metadata == nil` condition, I suppose there is little harm to clearing `Names` because over time any valid reference removed from `Names` will eventually return via the various reprocessing workflows.

I'll update this to clear Names.

Something we should consider longer term is to separate the image object from the name and names - similar to the typical image registry.  As names change (ie: `latest`) the id/name association is immediately stale (which I believe is the root of the concern here).  With that we could cleanup individual names as references change or that no longer have associated deployments, as well as store enrichment success/failure history per name (outside of logs) - so that we can present to the user and aid troubleshooting.